Automattic/woocommerce-payments
# Prevent form submission when clicking the buy with platform checkout button

#### Changes proposed in this Pull Request









Prior to this commit the entire checkout block form was submitted after the platform pay button was clicked, causing the entire page to reload.

#### Testing instructions

1. Try clicking the buy button with the **base branch** checked out. It should cause the browser to navigate to a different page.
2. Try the above but with this branch checked out. The browser shouldn't navigate away.

I am not sure if the bug happens on browsers other than Firefox, on setups working faster than mine when I was testing this... however the fact remains that a HTTP request is made when the button is clicked alongside the expected AJAX request. You could confirm this by going to your web developer tools' "Network" tab.





-------------------


**Post merge**
